kafka-impl/src/main/java/io/streamnative/pulsar/handlers/kop/KafkaTopicManager.java

```diff
@@ -157,10 +153,9 @@ public static void clearTopicManagerCache() {
         KOP_ADDRESS_CACHE.clear();
     }
 
-    // exception throw for pulsar.getClient();
-    private Producer registerInPersistentTopic(PersistentTopic persistentTopic) throws Exception {
+    private Producer registerInPersistentTopic(PersistentTopic persistentTopic) {
         Producer producer = new InternalProducer(persistentTopic, internalServerCnx,
-            ((PulsarClientImpl) (pulsarService.getClient())).newRequestId(),
+            KafkaProtocolHandler.getPulsarClientImpl(pulsarService).newRequestId(),
```

Comment:
If you create the pulsar client with listener name `kafkaListenerName`, this client should be only used to do topic lookup for kafka client, because the lookup result is kafka advertised address.
Here you create a pulsar producer from this client to send pulsar msg, that will casue pulsar producer connect to kafka advertised address and fail to send msg.

Reply:
I missunderstanded the code here, the `InternalProducer` will not do topic lookup. But I found `GroupCoordinator` seems to create a pulsar producer from this client, please also have a check.

Reply:
Fix it.